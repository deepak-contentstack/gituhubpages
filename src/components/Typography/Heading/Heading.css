@import '../../../../tokens/variables.scss';
@import '../../../../tokens/tokens.css';
@import '../../../static/styles/css/typography.css';
@import '../../../static/styles/css/variable.css';
@import '../../../static/styles/css/common.css';

h1,h2,h3,h4,h5,h6{
    font-family: $font-family-primary;
    color: $color-font-black;
    margin: 0;
}

h1{
    font-size: $size-font-6-xl;
    font-weight: $font-weight-bold;
}

h2{
    font-size: $size-font-5-xl;
    font-weight: $font-weight-bold;
}

h3{
    font-size: $size-font-4-xl;
    font-weight: $font-weight-extra-bold;
}

h4{
    font-size: $size-font-3-xl;
    font-weight: $font-weight-extra-bold;
}

h5{
    font-size: $size-font-2-xl;
    font-weight: $font-weight-extra-bold;
}

h6{
    font-size: $size-font-xl;
    font-weight: $font-weight-extra-bold;
}