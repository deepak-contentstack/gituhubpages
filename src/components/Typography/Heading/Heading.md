## **Heading**

The Heading component allows users to apply appropriate heading styles in their page/content.

## When to use 

Headings help users to read and understand text. They act as signposts for the readers and make it easier for them to figure out what a post or page is about. It also defines which parts of your content are important and shows how they're interconnected.

## Properties

### `text`
- Data that you want to display in tag

### `testId`
- Pass an ID that you can use for testing purposes. It is applied as a data attribute (data-test-id)

### `className`
- Pass the class names to be appended to this prop

### `tagName`
- Name of the <html> tag