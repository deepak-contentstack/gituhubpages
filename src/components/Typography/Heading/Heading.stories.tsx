import React from 'react'
import DocSubData from '../../../utils/components/DocSubData'
import Heading from '../Heading/Heading'
import notes from './Heading.md'

const tagArray: any = ['H1', 'H2', 'H3', 'H4', 'H5', 'H6']

const componentDescription = 'The Heading component allows users to apply appropriate heading styles in their page/content.'

const UsageDescription = () => {
    return (
        <>
            <p className="usage-description">Headings help users to read and understand text. They act as signposts for the readers and make it easier for them to figure out what a post or page is about. It also defines which parts of your content are important and shows how they're interconnected.</p>
            <p className="usage-description">H1-H6 are heading tags with varying range of height and font-weight that you can use as per your requirement.</p>
        </>)
}

export default {
    title: 'Components / Typography/Heading',
    component: Heading,
    argTypes: {
        tagName: {
            control: {
                type: "select",
                options: tagArray
            }
        },
        className: {
            control: 'disabled'
        },
    },
    parameters: {
        componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={<UsageDescription />} />,
        design: {
            type: 'figma',
            url: 'https://www.figma.com/proto/ovlEsic8nr81y5BOu1OlqD/ui-inventory-v2?node-id=451%3A0&scaling=min-zoom',
        }
    },
}

export const Default = (args) => {
    return (
        <>
            <Heading tagName={args.tagName} text={args.text} />
            <br />
            <Heading tagName="h2" text='Heading 2' />
            <br />
            <Heading tagName="h3" text='Heading 3' />
            <br />
            <Heading tagName="h4" text='Heading 4' />
            <br />
            <Heading tagName="h5" text='Heading 5' />
            <br />
            <Heading tagName="h6" text='Heading 6' />
        </>
    )
}
Default.args = {
    text: 'Heading 1',
    tagName: 'H1'
}

