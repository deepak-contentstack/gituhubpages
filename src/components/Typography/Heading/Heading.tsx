import React, { Component } from 'react'
import './Heading.css'

export type HeadingProps = {
  /**
  * Pass the class names to be appended to this prop
  */
  className?: string
  /**
  * Name of the html tag
  */
  tagName?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  /**
  * Data that you want to display in tag
  */
  text?: string
  /**
  * Pass an ID that you can use for testing purposes. It is applied as a data attribute (data-test-id)
  */
  testId?: string;
} & typeof defaultProps

const defaultProps = {
  tagName: 'h1',
  text: 'Heading 1',
  testId: 'cs-heading-tag'
}

export class Heading extends Component<HeadingProps> {
  static defaultProps = defaultProps

  render() {
    const { tagName, text, className, testId } = this.props
    const Element: any = tagName
    return (
      <Element className={className} data-test-id={testId}>
        {text}
      </Element>
    )
  }
}

export default Heading
