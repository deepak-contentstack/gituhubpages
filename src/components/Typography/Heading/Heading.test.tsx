import React from 'react'
import renderer from 'react-test-renderer'

import Heading from '../Heading/Heading'

describe('Heading', () => {
  it('H1 tag', () => {
    const H1 = () => {
      return (
        <Heading tagName='h1' text='Heading 1' />
      )
    }
    const tree = renderer.create(<H1 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('H2 tag', () => {
    const H2 = () => {
      return (
        <Heading tagName='h2' text='Heading 2' />
      )
    }
    const tree = renderer.create(<H2 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('H3 tag', () => {
    const H3 = () => {
      return (
        <Heading tagName='h3' text='Heading 3' />
      )
    }
    const tree = renderer.create(<H3 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('H4 tag', () => {
    const H4 = () => {
      return (
        <Heading tagName='h4' text='Heading 4' />
      )
    }
    const tree = renderer.create(<H4 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('H5 tag', () => {
    const H5 = () => {
      return (
        <Heading tagName='h5' text='Heading 5' />
      )
    }
    const tree = renderer.create(<H5 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('H6 tag', () => {
    const H6 = () => {
      return (
        <Heading tagName='h6' text='Heading 6' />
      )
    }
    const tree = renderer.create(<H6 />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component with an additional class name', () => {
    const H1 = () => {
      return (
        <Heading tagName="h1" className="heading-tag" text='Heading 1' />
      )
    }
    const tree = renderer.create(<H1 />).toJSON();
    expect(tree.props.className.includes("heading-tag")).toBe(true);
  })

})
