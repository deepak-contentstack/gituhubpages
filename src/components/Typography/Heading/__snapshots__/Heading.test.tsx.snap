// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading H1 tag 1`] = `
<h1
  data-test-id="cs-heading-tag"
>
  Heading 1
</h1>
`;

exports[`Heading H2 tag 1`] = `
<h2
  data-test-id="cs-heading-tag"
>
  Heading 2
</h2>
`;

exports[`Heading H3 tag 1`] = `
<h3
  data-test-id="cs-heading-tag"
>
  Heading 3
</h3>
`;

exports[`Heading H4 tag 1`] = `
<h4
  data-test-id="cs-heading-tag"
>
  Heading 4
</h4>
`;

exports[`Heading H5 tag 1`] = `
<h5
  data-test-id="cs-heading-tag"
>
  Heading 5
</h5>
`;

exports[`Heading H6 tag 1`] = `
<h6
  data-test-id="cs-heading-tag"
>
  Heading 6
</h6>
`;
